app/ide-desktop/lib/dashboard/src/authentication/src/http.tsx

```diff
@@ -0,0 +1,178 @@
+/** @file HTTP client definition that includes default HTTP headers for all sent requests.
+ *
+ * Used to build authenticated clients for external APIs, like our Cloud backend API. */
+
+
+
+// ==================
+// === HttpMethod ===
+// ==================
+
+/** HTTP method variants that can be used in an HTTP request. */
+enum HttpMethod {
+    get = "GET",
+    post = "POST",
+    put = "PUT",
+    delete = "DELETE",
+}
+
+
+
+// ============
+// === Json ===
+// ============
+
+/** A JavaScript object that can be serialized to a JSON value for an HTTP body. */
+type Json = Record<string, any>;
```

Comment:
it's a good idea to avoid `any` - this could probably be `Record<string, unknown>`

Reply:
Good to know, will change.

Reply:
yeah `any` explicitly suppresses type errors, instead of `unknown` which is just the base type, like `object` in Java